spring-cloud-circuitbreaker-resilience4j/src/main/java/org/springframework/cloud/circuitbreaker/resilience4j/Resilience4JAutoConfiguration.java

```diff
@@ -85,6 +86,17 @@ public Resilience4jBulkheadProvider bulkheadProvider(ThreadPoolBulkheadRegistry
 
 	}
 
+	@Configuration(proxyBeanMethods = false)
+	@ConditionalOnBean({ MeterRegistry.class })
+	public static class MicrometerResilience4JGroupCustomizerConfiguration {
+
+		@Bean
+		MeterRegistryCustomizer<MeterRegistry> metricsCommonTags() {
+			return registry -> registry.config().commonTags(Resilience4JCircuitBreaker.CIRCUIT_BREAKER_GROUP_TAG, "none");
```

Comment:
This way the tag will be added for every single meter you have (e.g.: Spring MVC metrics, Kafka metrics, etc). Is it what you want or do you want to only add this tag to Resilience4J-related metrics?

We usually recommend adding fallback tag values at the instrumentation level, e.g.:
```
Timer timer = Timer.builder("my.timer")
    .tags("group", circuitBreaker.getGroupName() != null ? circuitBreaker.getGroupName() : "none")
```

Here's a real-life example when we attach the name of the exception as a tag but we set value "none" if there were no exceptions: [TimedAspect.java#L216-L227](https://github.com/micrometer-metrics/micrometer/blob/708c62def62c477700a22e15145be689af95fbdb/micrometer-core/src/main/java/io/micrometer/core/aop/TimedAspect.java#L216-L227). I guess your use-case is similar.

If you have no control over this, I recommend using a `MeterFilter` where you can selectively attach the missing tag based on the name and the current tags of the Meter.

Reply:
> This way the tag will be added for every single meter you have (e.g.: Spring MVC metrics, Kafka metrics, etc). Is it what you want or do you want to only add this tag to Resilience4J-related metrics?




No just resilience4j metrics, is there a better way to do that?





> If you have no control over this, I recommend using a MeterFilter where you can selectively attach the missing tag based on the name and the current tags of the Meter.

I think the problem is here https://github.com/resilience4j/resilience4j/blob/master/resilience4j-spring/src/main/java/io/github/resilience4j/bulkhead/configure/threadpool/ThreadPoolBulkheadConfiguration.java#L72-L74
Which we don't have control over.  They are not using the tags like we are.

Reply:
I would suggest creating a `MeterFilter` (boot will auto-config it for you), like this: https://micrometer.io/docs/concepts#_transforming_metrics
So you can check if this is a resilience4j-related meter and the group tag is missing, add it with a default value.